sycl/test/basic_tests/stdcpp_compat.cpp

```diff
@@ -10,13 +9,15 @@
 
 #include <sycl/sycl.hpp>
 
-// cxx14-warning@* {{DPCPP does not support C++ version earlier than C++17. Some features might not be available.}}
+// clang-format off
+// cxx14-error@* {{static assertion failed due to requirement '201402L >= 201703L'}}
```

Comment:
I don't think it works reliably - any attempt to use C++17 feature without macro guard would result in an extra error reported and the test would fail.

Reply:
Hm, the `static_assert` should fire very early as it's in the most basic header. But to make the test more reliable, probably, we could add `-Xclang -verify-ignore-unexpected=error,note` and  `-Xclang -ferror-limit=<Big number>` options.